modules/andes-core/broker/src/main/java/org/dna/mqtt/moquette/messaging/spi/impl/ProtocolProcessor.java

```diff
@@ -688,19 +757,21 @@ void processSubscribe(ServerChannel session, SubscribeMessage msg, String client
             }
         }
 
-        //ack the client
-        SubAckMessage ackMessage = new SubAckMessage();
-        ackMessage.setMessageID(msg.getMessageID());
+        // Ack the client only if all subscriptions are successful
+        if (authenticated) {
```

Comment:
When a single subscription in the subscriptions list fails this flag will be false, resulting in not sending a SubAckMessage. Shouldn't we send a SubAckMessage if at least one subscription is successful?